python/pyspark/mllib/linalg.py

```diff
@@ -1152,9 +1156,385 @@ def sparse(numRows, numCols, colPtrs, rowIndices, values):
         return SparseMatrix(numRows, numCols, colPtrs, rowIndices, values)
 
 
+class DistributedMatrix(object):
+    """Represents a distributively stored matrix backed by one or more RDDs."""
+    def numRows(self):
+        """Get or compute the number of rows."""
+        raise NotImplementedError
+
+    def numCols(self):
+        """Get or compute the number of cols."""
+        raise NotImplementedError
+
+
+class DistributedMatrices(object):
+    """Factory methods for distributed matrices."""
+    @staticmethod
+    def rowMatrix(rows, numRows=0, numCols=0):
+        """
+        Create a RowMatrix.
+
+        :param rows: An RDD of Vectors.
+        """
+        javaRowMatrix = callMLlibFunc("createRowMatrix", rows, long(numRows), int(numCols))
```

Comment:
Great, it will take minimal amount of effort to just move this over to the class `RowMatrix`